### PR TITLE
Add labels to PR stages

### PR DESCRIPTION
**Pull request by 0xc0170**
+ define what is a component

Questions:

- regarding components: we might want to list all of available or would it be clear from Mbed OS structure?
- I defined "ready for merge" as: does not mean it will be instantly merged, expecting to be checked during gatekeeping but most cases (simple PR) can be merged immediately - how it is
- we are still missing here  the information about our CI @OPpuolitaival I would like your help here to add a section here to add `CI` and provide info about public information that we have on each pull request (separate PR that would address this)

cc @SeppoTakalo @AnttiKauppila @ARMmbed/mbed-os-maintainers

### CODE REVIEW

**Comment by SeppoTakalo** [comment]
I don't understand the usecase of component label, so I would even prefer to leave it out.

**Comment by 0xc0170** [comment]
> I don't understand the usecase of component label, so I would even prefer to leave it out.

It's not for PR but only for issues to be able to sort them somehow (visible here https://github.com/ARMmbed/mbed-os/issues ). The support team is taking over the issues soon, we can discuss with them to gather feedback about the component label and its usability.

**Comment by OPpuolitaival** [comment]
@0xc0170 I will add the CI information a bit later

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -194,7 +204,7 @@ We use many other labels to summarize the scope and effect of the changes.
 - *needs: preceding PR* - This pull request cannot yet be merged because it has a dependency on another pull request that needs to merge first.
 - *DO NOT MERGE* - This pull request contains changes that may be in a draft state and submitted purely for review, or may contain breaking changes that have not been considered.
 - *devices: 'name'* - The pull request specifically affects the named device(s).
-- *component: 'name'* - The pull request specifically affects the named component.
+- *component: 'name'* - The pull request specifically affects the named component. Component names follow the structure of Mbed OS (`ble`, `storage`, `tls` and so on).
```
Comment (AnotherButler) [inline docs/reference/contributing/guidelines/workflow.md]: Query: When we say component here, do we mean the new components, such as the new storage ones?

Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: All the components we have (Mbed OS directory structure to be specific) such as ble, storage, drivers, hal , etc

Comment (cmonr) [reply docs/reference/contributing/guidelines/workflow.md]: @sg- Any suggestions?

**Comment by 0xc0170** [comment]
I think this is ready. The support team will review component label and its usage. This covers the current situation.

Ready 🙌

**Pull request merged**